File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    new_s = []

    for i in s:
        new_s.append(i)
        if(len(new_s) == 0 or len(new_s) == 1): pass
        elif(i ==')'):
            new_s.pop()
            new_s.pop()
    if(len(new_s) == 0): return True
    return False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        boolean answer = false;
        int stack = 0;
        for(char n: s.toCharArray()){
            stack += 1;
            if(stack == 1) continue;
            else if(n == ')'){
                stack -= 2;
            }
        }
        if(stack == 0) return true;
        return answer;
    }
}